### DISCUSSION TITLE
Compiling tasmota for Olimex ESP32-POE board

### DISCUSSION THREAD

**Discussion by dexterbot80** [arendst/Tasmota#22522] [General]
Hello, 
I would like to compile a custom firmware for the OLIMEX ESP32-POE board.
By default, the esp8266 board is selected in platformio.ini
Can someone help me to configure correctly for ESP32-POE?
Thank you

**Comment by s-hadinger** [comment]
The standard ESP32 build works fine with ESP32-POE

**Reply by dexterbot80** [reply]
> The standard ESP32 build works fine with ESP32-POE

I understand but i need  an thermostat with PID function and is not activated in standard builds

**Comment by dexterbot80** [comment]
Does anyone know how to do it?
Thank you

**Reply by barbudor** [reply]
Have you started from the docs ? https://tasmota.github.io/docs/Compile-your-build/

The idea is that you have to customize your own build of tasmota32 by adding features through `#define`  in the `user_config_override.h`

Don't forget that you can only customize the base binary of Tasmota (tasmota for esp8266, tasmota32 for esp32, tasmota32c3 for esp32c3 etc ...) and not any of the variants

**Reply by s-hadinger** [reply]
À good start is reading this: https://tasmota.github.io/docs/Compile-your-build/
